spec.go

```diff
@@ -14,7 +16,32 @@ type ExistsConfig struct {
 	Organizations []string
 }
 
+type SearchConfig struct {
+	ID            string
+	Organizations []string
+}
+
+type UpdateConfig struct {
+	AllowBareDomains bool
+	AllowSubdomains  bool
+	AltNames         []string
+	ID               string
+	Organizations    []string
+	TTL              string
+}
+
 type Interface interface {
 	Create(config CreateConfig) error
 	Exists(config ExistsConfig) (bool, error)
+	Search(config SearchConfig) (Role, error)
+	Update(config UpdateConfig) error
+}
```

Comment:
The interface is now ready to support a better reconciliation of the Vault role. I think in the future we may want to use the `Role` type more. I want to keep the PR down to the fundamentals though. It is already a bit much in terms of different changes.